### PR TITLE
Serial generation time in db

### PR DESCRIPTION
**Pull request by lolepezy**
This is to introduce the 'catch up' mechanism for deltas.

If there is more than one instance of a publication server, a situation when one of them 'freezes' the version of the DB and the others don't know about it will be common. This PR introduces correct handling of this type of situations. `updateFS` will update its RRDP/rsync repository with all serials between the last one it remembers and the one in the DB.

### CODE REVIEW

File: src/main/scala/net/ripe/rpki/publicationserver/repository/DataFlusher.scala [hunk 1]
```diff
@@ -213,13 +236,26 @@ class DataFlusher(conf: AppConfig)(implicit val system: ActorSystem)
...
+//    val stream = createStream
+//    try {
+//      f(stream)
+//      stream.summary
+//    } finally {
+//      stream.close()
+//    }
+//  }
+
+  def withTmpStream(targetFile: Path)(f : HashingSizedStream => Unit) = {
```
Comment (ties) [inline src/main/scala/net/ripe/rpki/publicationserver/repository/DataFlusher.scala:249]: `withSwapAfterCloseStream`?

**Review by ties — APPROVED** [review]
Looks good to me. Have not tested it.

File: src/main/scala/net/ripe/rpki/publicationserver/repository/DataFlusher.scala (hunk 1 above)
Comment (lolepezy) [reply src/main/scala/net/ripe/rpki/publicationserver/repository/DataFlusher.scala:239]: withAtomicStream?

Comment (ties) [reply src/main/scala/net/ripe/rpki/publicationserver/repository/DataFlusher.scala]: Yeap!

**Pull request merged**